### PR TITLE
Print succinct output for runtime statistics

### PR DESCRIPTION
**Pull request by majetideepak**
The runtime statistics that are optionally printed as part of `printPlanWithStats` have been extended to print succinct output.
Sample output is below.
```
-> Project[expressions: (c0:INTEGER, ROW["c0"]), (p1:BIGINT, plus(ROW["c1"],1)), (p2:BIGINT, plus(ROW["c1"],ROW["u_c1"]))]
   Output: 2000 rows (154.98KB), Cpu time: 4.23ms, Blocked wall time: 0ns, Peak memory: 1.00MB, Threads: 1
      dataSourceLazyWallNanos    sum: 1.18ms, count: 20, min: 42.00us, max: 116.00us
  -> HashJoin[INNER c0=u_c0]
     Output: 2000 rows (136.88KB), Cpu time: 1.71ms, Blocked wall time: 434.00us, Peak memory: 2.00MB
     HashBuild: Input: 100 rows (1.31KB), Output: 0 rows (0B), Cpu time: 394.40us, Blocked wall time: 0ns, Peak memory: 1.00MB, Threads: 1
        distinctKey0       sum: 101, count: 1, min: 101, max: 101
        queuedWallNanos    sum: 98.00us, count: 1, min: 98.00us, max: 98.00us
        rangeKey0          sum: 200, count: 1, min: 200, max: 200
     HashProbe: Input: 2000 rows (118.12KB), Output: 2000 rows (136.88KB), Cpu time: 1.32ms, Blocked wall time: 434.00us, Peak memory: 1.00MB, Threads: 1
        dynamicFiltersProduced    sum: 1, count: 1, min: 1, max: 1
        queuedWallNanos           sum: 27.00us, count: 1, min: 27.00us, max: 27.00us
    -> TableScan[Table: projection]
       Input: 2000 rows (118.12KB), Raw Input: 20480 rows (72.31KB), Output: 2000 rows (118.12KB), Cpu time: 19.04ms, Blocked wall time: 14.00us, Peak memory: 1.00MB, Threads: 1, Splits: 20
          dataSourceWallNanos       sum: 8.73ms, count: 40, min: 46.00us, max: 979.00us
          dynamicFiltersAccepted    sum: 1, count: 1, min: 1, max: 1
          localReadBytes            sum: 0B, count: 1, min: 0B, max: 0B
          numLocalRead              sum: 0, count: 1, min: 0, max: 0
          numPrefetch               sum: 36, count: 1, min: 36, max: 36
          numRamRead                sum: 0, count: 1, min: 0, max: 0
          numStorageRead            sum: 140, count: 1, min: 140, max: 140
          prefetchBytes             sum: 54.00KB, count: 1, min: 54.00KB, max: 54.00KB
          queuedWallNanos           sum: 45.00us, count: 1, min: 45.00us, max: 45.00us
          ramReadBytes              sum: 0B, count: 1, min: 0B, max: 0B
          skippedSplitBytes         sum: 0B, count: 1, min: 0B, max: 0B
          skippedSplits             sum: 0, count: 1, min: 0, max: 0
          skippedStrides            sum: 0, count: 1, min: 0, max: 0
          storageReadBytes          sum: 150.25KB, count: 1, min: 150.25KB, max: 150.25KB
    -> Project[expressions: (u_c0:INTEGER, ROW["c0"]), (u_c1:BIGINT, ROW["c1"])]
       Output: 100 rows (1.31KB), Cpu time: 69.04us, Blocked wall time: 0ns, Peak memory: 0B, Threads: 1
      -> Values[100 rows in 1 vectors]
         Input: 0 rows (0B), Output: 100 rows (1.31KB), Cpu time: 6.01us, Blocked wall time: 0ns, Peak memory: 0B, Threads: 1
```

### CODE REVIEW

**Review by mbasmanova — COMMENTED** [review]
@majetideepak This is a nice change. I have some comments on the design of the structures.

**Comment by majetideepak** [comment]
@mbasmanova good points on the API. I made some changes. Can you take another look? Thanks.

File: velox/common/base/RuntimeMetrics.h [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+
+#pragma once
+
+#include <sstream>
+
+namespace facebook::velox {
+
+struct RuntimeCounter {
+  enum Unit { kNone, kNanos, kBytes };
+  int64_t value{0};
```
Comment (mbasmanova) [inline velox/common/base/RuntimeMetrics.h:25]: Since there is no default constructor, the default values here are redundant.

File: velox/common/base/RuntimeMetrics.h [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+  int64_t count{0};
+  int64_t min{std::numeric_limits<int64_t>::max()};
+  int64_t max{std::numeric_limits<int64_t>::min()};
+
+  explicit RuntimeMetric(RuntimeCounter::Unit _unit = RuntimeCounter::kNone)
+      : unit(_unit) {}
+
+  void addValue(int64_t value);
+
+  void printMetric(std::stringstream& stream) const;
```
Comment (mbasmanova) [inline velox/common/base/RuntimeMetrics.h:45]: nit: toString() might be more conventional name

File: velox/common/base/RuntimeMetrics.h [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+ */
+
+#pragma once
+
+#include <sstream>
+
+namespace facebook::velox {
+
+struct RuntimeCounter {
+  enum Unit { kNone, kNanos, kBytes };
```
Comment (mbasmanova) [inline velox/common/base/RuntimeMetrics.h:24]: use enum class

File: velox/exec/tests/utils/PlanBuilder.h [hunk 4]
```diff
@@ -54,7 +54,9 @@ class PlanBuilder {
   explicit PlanBuilder(memory::MemoryPool* pool = nullptr)
       : PlanBuilder(std::make_shared<PlanNodeIdGenerator>(), pool) {}
 
-  PlanBuilder& tableScan(const RowTypePtr& outputType);
```
Comment (mbasmanova) [inline velox/exec/tests/utils/PlanBuilder.h:57]: This is a nice change, but feels unrelated. Would it make sense to move it into a separate PR?

File: velox/exec/Operator.cpp [hunk 5]
```diff
@@ -323,8 +323,11 @@ void OperatorStats::add(const OperatorStats& other) {
 
   memoryStats.add(other.memoryStats);
 
-  for (const auto& stat : other.runtimeStats) {
-    runtimeStats[stat.first].merge(stat.second);
+  for (const auto& [name, stats] : other.runtimeStats) {
+    if (UNLIKELY(runtimeStats.count(name) == 0)) {
+      runtimeStats.insert(std::make_pair(name, RuntimeMetric(stats.unit)));
```
Comment (mbasmanova) [inline velox/exec/Operator.cpp:328]: Do you need to create new RuntimeMetric instance? Can you instead use 'stats' ? then move the code on 330 into the 'else' clause?

**Review by mbasmanova — COMMENTED** [review]
@majetideepak Looks good. Some additional comments.

File: velox/common/base/RuntimeMetrics.h (hunk 2 above)
Comment (majetideepak) [reply velox/common/base/RuntimeMetrics.h:45]: I felt `toString()` API usually returns a `string` `std::string toString() const`. I sort of followed `printCustomStats` here. I can use `toString()` if you prefer.

**Comment by majetideepak** [comment]
@mbasmanova I realized we need to change the metric names as well. For example, I changed `queuedWallNanos->ueuedWallTime`

**Comment by mbasmanova** [comment]
@majetideepak 

> I realized we need to change the metric names as well.

What's the motivation for this change?

File: velox/exec/Driver.cpp [hunk 6]
```diff
@@ -262,7 +265,8 @@ StopReason Driver::runInternal(
   // been deleted.
   if (curOpIndex_ < operators_.size()) {
     operators_[curOpIndex_]->stats().addRuntimeStat(
-        "queuedWallNanos", queuedTime);
+        "queuedWallTime",
```
[the remark this answers is not in the document]
Comment (majetideepak) [reply velox/exec/Driver.cpp:268]: sounds good. Will revert the rename.

**Comment by majetideepak** [comment]
> What's the motivation for this change?

Just that for example `queuedWallNanos  sum: 1s` looks odd as the value is in seconds but the name contains `Nanos`. I will revert this for now as discussed above.

**Comment by mbasmanova** [comment]
> Just that for example queuedWallNanos sum: 1s looks odd as the value is in seconds but the name contains Nanos. 

Got it. Thanks for clarifying.

> I will revert this for now as discussed above.

Thanks.

**Review by mbasmanova — APPROVED** [review]
@majetideepak Thanks.

**Comment by mbasmanova** [comment]
@majetideepak This PR changes the Connector interface, hence, requires updating all the internal connectors. I'm working on that, hence, the delay in landing.

**Comment by majetideepak** [comment]
@mbasmanova thanks for the update and help with this.
